### ISSUE TITLE
mount.fstab: /usr/local/appjail/jails/brave/jail//dev/shm: No such file or directory

### ISSUE THREAD

**Issue by kenrap** [AppJail-makejails/brave#1]
Hello,

Forgive me if I'm not installing Brave properly here. I got this error which doesn't make sense to me since the directory path does actually exist for me.

Here are my system details:
```
# pkg info --origin appjail-devel           
appjail-devel-g20230628        sysutils/appjail-devel
# uname -v
FreeBSD 14.0-CURRENT #0 main-n263872-24ec316c13bd: Thu Jun 29 05:58:23 PDT 2023     root@freebsd:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG
```

Originally, I tried running `appjail makejail -j brave -- --ruleset 13` but I ended up with this error.

```
[00:00:02] [ error ] [brave] --ruleset: Invalid option.
```

And then I tried running without the `--ruleset` flag which gave me this confusing error near the end of this output:

```
# appjail makejail -j brave                
[00:00:00] [ info  ] [brave] Building brave ...
[00:00:00] [ debug ] [brave] Main Makejail: Makejail
[00:00:00] [ debug ] [brave] Using method:file (args:Makejail) from Makejail.
[00:00:00] [ debug ] [brave] Including /root/brave/Makejail ...
[00:00:00] [ debug ] [brave] Using method:file (args:options/image-options.makejail) from options/image-options.makejail.
[00:00:00] [ debug ] [brave] Including /root/brave/options/image-options.makejail ...
[00:00:00] [ debug ] [brave] Using method:file (args:options.makejail) from options.makejail.
[00:00:00] [ debug ] [brave] Including /root/brave/options/options.makejail ...
[00:00:01] [ debug ] [brave] Using method:file (args:pkgs-host.makejail) from pkgs-host.makejail.
[00:00:01] [ debug ] [brave] Including /root/brave/pkgs-host.makejail ...
[00:00:01] [ debug ] [brave] Using method:file (args:hosts.makejail) from hosts.makejail.
[00:00:01] [ debug ] [brave] Including /root/brave/hosts.makejail ...
[00:00:01] [ debug ] [brave] Makejail generated:
[00:00:01] [ debug ] [brave]     RAW cd -- "/root/brave/options" # Makejail: /root/brave/options/image-options.makejail
[00:00:01] [ debug ] [brave]     OPTION tmpdir
[00:00:01] [ debug ] [brave]     OPTION linuxfs
[00:00:01] [ debug ] [brave]     OPTION resolv_conf
[00:00:01] [ debug ] [brave]     OPTION tzdata
[00:00:01] [ debug ] [brave]     OPTION overwrite
[00:00:01] [ debug ] [brave]     OPTION start
[00:00:01] [ debug ] [brave]     RAW cd -- "/root/brave" # Makejail: /root/brave/Makejail
[00:00:01] [ debug ] [brave]     ARG brave_tag=latest
[00:00:01] [ debug ] [brave]     ARG brave_display=:0
[00:00:01] [ debug ] [brave]     ARG brave_pulse_server=unix:/tmp/pulse-native
[00:00:01] [ debug ] [brave]     FROM --entrypoint gh+AppJail-makejails/brave brave:${brave_tag}
[00:00:01] [ debug ] [brave]     PKG --local pulseaudio
[00:00:01] [ debug ] [brave]     COPY etc
[00:00:01] [ debug ] [brave]     REPLACE /etc/hosts HOSTNAME "${APPJAIL_JAILNAME}${HOST_DOMAIN}"
[00:00:01] [ debug ] [brave]     CMD --local mkdir -p /usr/local/bin
[00:00:01] [ debug ] [brave]     CMD --local install -m 555 files/brave-appjail.in /usr/local/bin/brave-appjail
[00:00:01] [ debug ] [brave]     CMD --local sed -i '' -Ee "s/%%JAILNAME%%/${APPJAIL_JAILNAME}/" /usr/local/bin/brave-appjail
[00:00:01] [ debug ] [brave]     CMD --local sed -i '' -Ee "s/%%DISPLAY%%/${brave_display}/" /usr/local/bin/brave-appjail
[00:00:01] [ debug ] [brave]     CMD --local sed -i '' -Ee "s#%%PULSE_SERVER%%#${brave_pulse_server}#" /usr/local/bin/brave-appjail
[00:00:01] [ debug ] [brave]     RAW for d in 48 128 16 24 256 32 64; do
[00:00:01] [ debug ] [brave]     RAW mkdir -p /usr/local/share/icons/hicolor/${d}x${d}/apps
[00:00:01] [ debug ] [brave]     RAW cp ${APPJAIL_JAILDIR}/usr/share/icons/hicolor/${d}x${d}/apps/brave-browser.png /usr/local/share/icons/hicolor/${d}x${d}/apps
[00:00:01] [ debug ] [brave]     RAW done
[00:00:01] [ debug ] [brave]     CMD --local mkdir -p /usr/local/share/applications
[00:00:01] [ debug ] [brave]     CMD --local cp files/brave-appjail.desktop /usr/local/share/applications
[00:00:01] [ debug ] [brave]     CMD --local mkdir -p /usr/local/etc/rc.d
[00:00:01] [ debug ] [brave]     CMD --local cp files/pulseaudio.in /usr/local/etc/rc.d/pulseaudio
[00:00:01] [ debug ] [brave]     CMD --local chmod +x /usr/local/etc/rc.d/pulseaudio
[00:00:01] [ debug ] [brave]     STOP
[00:00:01] [ debug ] [brave]     STAGE create
[00:00:01] [ debug ] [brave]     CMD --local xhost +
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/RAW (args:cd -- "/root/brave/options" # Makejail: /root/brave/options/image-options.makejail)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/build/OPTION (args:tmpdir)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/build/OPTION (args:linuxfs)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/build/OPTION (args:resolv_conf)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/build/OPTION (args:tzdata)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/build/OPTION (args:overwrite)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/build/OPTION (args:start)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/RAW (args:cd -- "/root/brave" # Makejail: /root/brave/Makejail)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/ARG (args:brave_tag=latest)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/ARG (args:brave_display=:0)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/ARG (args:brave_pulse_server=unix:/tmp/pulse-native)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/build/FROM (args:--entrypoint gh+AppJail-makejails/brave brave:${brave_tag})
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/PKG (args:--local pulseaudio)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/COPY (args:etc)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/REPLACE (args:/etc/hosts HOSTNAME "${APPJAIL_JAILNAME}${HOST_DOMAIN}")
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local mkdir -p /usr/local/bin)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local install -m 555 files/brave-appjail.in /usr/local/bin/brave-appjail)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local sed -i '' -Ee "s/%%JAILNAME%%/${APPJAIL_JAILNAME}/" /usr/local/bin/brave-appjail)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local sed -i '' -Ee "s/%%DISPLAY%%/${brave_display}/" /usr/local/bin/brave-appjail)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local sed -i '' -Ee "s#%%PULSE_SERVER%%#${brave_pulse_server}#" /usr/local/bin/brave-appjail)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/RAW (args:for d in 48 128 16 24 256 32 64; do)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/RAW (args:mkdir -p /usr/local/share/icons/hicolor/${d}x${d}/apps)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/RAW (args:cp ${APPJAIL_JAILDIR}/usr/share/icons/hicolor/${d}x${d}/apps/brave-browser.png /usr/local/share/icons/hicolor/${d}x${d}/apps)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/RAW (args:done)
[00:00:01] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local mkdir -p /usr/local/share/applications)
[00:00:02] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local cp files/brave-appjail.desktop /usr/local/share/applications)
[00:00:02] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local mkdir -p /usr/local/etc/rc.d)
[00:00:02] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local cp files/pulseaudio.in /usr/local/etc/rc.d/pulseaudio)
[00:00:02] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local chmod +x /usr/local/etc/rc.d/pulseaudio)
[00:00:02] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/build/STOP (args:)
[00:00:02] [ debug ] [brave] Running makejail command (cmd): STAGE (args:create)
[00:00:02] [ debug ] [brave] Running makejail command (cmd): /usr/local/share/appjail/makejail/cmd/all/CMD (args:--local xhost +)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/ARG (input:/usr/local/appjail/cache/tmp/.appjail/appjail.DIaL2bzLEO)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/build/FROM (input:/usr/local/appjail/cache/tmp/.appjail/appjail.RS7i1VTlt4)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/build/OPTION (input:/usr/local/appjail/cache/tmp/.appjail/appjail.UqiNzUtHhE)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/RAW (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/0.RAW)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/RAW (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/7.RAW)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/PKG (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/12.PKG)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/COPY (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/13.COPY)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/REPLACE (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/14.REPLACE)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/15.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/16.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/17.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/18.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/19.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/RAW (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/20.RAW)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/RAW (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/21.RAW)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/RAW (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/22.RAW)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/RAW (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/23.RAW)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/24.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/25.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/26.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/27.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/all/CMD (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/28.CMD)
[00:00:02] [ debug ] [brave] Running makejail command (write): /usr/local/share/appjail/makejail/write/build/STOP (input:/usr/local/appjail/cache/tmp/.appjail/appjail.lMAl3fHqsd/stages/build/29.STOP)
[00:00:02] [ debug ] [brave] Buildscript generated:
[00:00:02] [ debug ] [brave]     . "${APPJAIL_CONFIG}"
[00:00:02] [ debug ] [brave]     . "${LIBDIR}/load"
[00:00:02] [ debug ] [brave]     
[00:00:02] [ debug ] [brave]     lib_load "${LIBDIR}/sysexits"
[00:00:02] [ debug ] [brave]     lib_load "${LIBDIR}/atexit"
[00:00:02] [ debug ] [brave]     lib_load "${LIBDIR}/log"
[00:00:02] [ debug ] [brave]     lib_load "${LIBDIR}/check_func"
[00:00:02] [ debug ] [brave]     
[00:00:02] [ debug ] [brave]     lib_atexit_init
[00:00:02] [ debug ] [brave]     set -e
[00:00:02] [ debug ] [brave]     brave_pulse_server="unix:/tmp/pulse-native"
[00:00:02] [ debug ] [brave]     brave_tag="latest"
[00:00:02] [ debug ] [brave]     brave_display=":0"
[00:00:02] [ debug ] [brave]     lib_load "${LIBDIR}/check_func"
[00:00:02] [ debug ] [brave]     
[00:00:02] [ debug ] [brave]     while [ $# -gt 0 ]; do
[00:00:02] [ debug ] [brave]            case "$1" in
[00:00:02] [ debug ] [brave]                    --brave_pulse_server)
[00:00:02] [ debug ] [brave]                            brave_pulse_server="$2"; shift
[00:00:02] [ debug ] [brave]                            ;;
[00:00:02] [ debug ] [brave]                    --brave_tag)
[00:00:02] [ debug ] [brave]                            brave_tag="$2"; shift
[00:00:02] [ debug ] [brave]                            ;;
[00:00:02] [ debug ] [brave]                    --brave_display)
[00:00:02] [ debug ] [brave]                            brave_display="$2"; shift
[00:00:02] [ debug ] [brave]                            ;;
[00:00:02] [ debug ] [brave]                    --)
[00:00:02] [ debug ] [brave]                            shift
[00:00:02] [ debug ] [brave]                            break
[00:00:02] [ debug ] [brave]                            ;;
[00:00:02] [ debug ] [brave]                    --*)
[00:00:02] [ debug ] [brave]                            lib_err ${EX_USAGE} -- "$1: Invalid option."
[00:00:02] [ debug ] [brave]                            ;;
[00:00:02] [ debug ] [brave]                    *)
[00:00:02] [ debug ] [brave]                            break
[00:00:02] [ debug ] [brave]                            ;;
[00:00:02] [ debug ] [brave]            esac
[00:00:02] [ debug ] [brave]     
[00:00:02] [ debug ] [brave]            shift
[00:00:02] [ debug ] [brave]     done
[00:00:02] [ debug ] [brave]     if lib_check_empty "$brave_pulse_server"; then
[00:00:02] [ debug ] [brave]            lib_err ${EX_DATAERR} "option requires an argument -- brave_pulse_server"
[00:00:02] [ debug ] [brave]     fi
[00:00:02] [ debug ] [brave]     if lib_check_empty "$brave_tag"; then
[00:00:02] [ debug ] [brave]            lib_err ${EX_DATAERR} "option requires an argument -- brave_tag"
[00:00:02] [ debug ] [brave]     fi
[00:00:02] [ debug ] [brave]     if lib_check_empty "$brave_display"; then
[00:00:02] [ debug ] [brave]            lib_err ${EX_DATAERR} "option requires an argument -- brave_display"
[00:00:02] [ debug ] [brave]     fi
[00:00:02] [ debug ] [brave]     "${APPJAIL_SCRIPT}" image import -a "amd64" -N .ajspec -n "brave" -t "${brave_tag}" -- "gh+AppJail-makejails/brave"
[00:00:02] [ debug ] [brave]     "${APPJAIL_SCRIPT}" image jail -a "amd64" -i "brave" -t "${brave_tag}" -- "${APPJAIL_JAILNAME}" "tmpdir" "linuxfs" "resolv_conf"  "tzdata" "overwrite" "start" 
[00:00:02] [ debug ] [brave]     cd -- "/root/brave/options" # Makejail: /root/brave/options/image-options.makejail
[00:00:02] [ debug ] [brave]     cd -- "/root/brave" # Makejail: /root/brave/Makejail
[00:00:02] [ debug ] [brave]     pkg install -y -- "pulseaudio"
[00:00:02] [ debug ] [brave]     cp  -a -- "etc" "${APPJAIL_JAILDIR}/"
[00:00:02] [ debug ] [brave]     lib_load "${LIBDIR}/replace"
[00:00:02] [ debug ] [brave]     
[00:00:02] [ debug ] [brave]     lib_mk_replace_var "${APPJAIL_JAILDIR}//etc/hosts" "HOSTNAME" "${APPJAIL_JAILNAME}${HOST_DOMAIN}" "${APPJAIL_JAILDIR}//etc/hosts"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "mkdir -p /usr/local/bin"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "install -m 555 files/brave-appjail.in /usr/local/bin/brave-appjail"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "sed -i '' -Ee \"s/%%JAILNAME%%/\${APPJAIL_JAILNAME}/\" /usr/local/bin/brave-appjail"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "sed -i '' -Ee \"s/%%DISPLAY%%/\${brave_display}/\" /usr/local/bin/brave-appjail"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "sed -i '' -Ee \"s#%%PULSE_SERVER%%#\${brave_pulse_server}#\" /usr/local/bin/brave-appjail"
[00:00:02] [ debug ] [brave]     for d in 48 128 16 24 256 32 64; do
[00:00:02] [ debug ] [brave]     mkdir -p /usr/local/share/icons/hicolor/${d}x${d}/apps
[00:00:02] [ debug ] [brave]     cp ${APPJAIL_JAILDIR}/usr/share/icons/hicolor/${d}x${d}/apps/brave-browser.png /usr/local/share/icons/hicolor/${d}x${d}/apps
[00:00:02] [ debug ] [brave]     done
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "mkdir -p /usr/local/share/applications"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "cp files/brave-appjail.desktop /usr/local/share/applications"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "mkdir -p /usr/local/etc/rc.d"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "cp files/pulseaudio.in /usr/local/etc/rc.d/pulseaudio"
[00:00:02] [ debug ] [brave]      env "brave_tag=${brave_tag}" "brave_display=${brave_display}" "brave_pulse_server=${brave_pulse_server}"  sh -c "chmod +x /usr/local/etc/rc.d/pulseaudio"
[00:00:02] [ debug ] [brave]     "${APPJAIL_SCRIPT}" stop -- "${APPJAIL_JAILNAME}"
[00:00:02] [ debug ] [brave] Cloning https://github.com/AppJail-makejails/brave as /usr/local/appjail/cache/tmp/.appjail/appjail.W6DJpX4xvw ...
[00:00:03] [ debug ] [brave] Fetching brave from https://dtxdf.duckdns.org/G0Y9r7zdzd/brave_latest_amd64_image_appjail: fetch -Rpm -o "/usr/local/appjail/cache/images/brave/latest-amd64-image.appjail" "https://dtxdf.duckdns.org/G0Y9r7zdzd/brave_latest_amd64_image_appjail"
/usr/local/appjail/cache/images/brave/latest-a         226 MB  974 kBps 03m58s
[00:04:06] [ info  ] [brave] Saved as /usr/local/appjail/cache/images/brave/latest-amd64-image.appjail
[00:04:06] [ debug ] [brave] quick parameters: import+root="input:/usr/local/appjail/cache/images/brave/latest-amd64-image.appjail" portable tmpdir linuxfs resolv_conf tzdata overwrite start
[00:04:06] [ info  ] [brave] Importing brave ...
[00:04:06] [ info  ] [brave] Creating an empty jail ...
[00:04:17] [ info  ] [brave] Done.
[00:04:17] [ debug ] [brave] Copying /etc/localtime as /usr/local/appjail/jails/brave/jail/etc/localtime
[00:04:17] [ debug ] [brave] Copying /etc/resolv.conf as /usr/local/appjail/jails/brave/jail/etc/resolv.conf
[00:04:17] [ debug ] [brave] Using tmpdir option ...
[00:04:17] [ debug ] [brave] Using linuxfs option ...
[00:04:18] [ debug ] [brave] Setting the boot flag to the brave jail ...
[00:04:18] [ debug ] [brave] Template generated:
[00:04:18] [ debug ] [brave]     exec.start: "/bin/sh /etc/rc"
[00:04:18] [ debug ] [brave]     exec.stop: "/bin/sh /etc/rc.shutdown jail"
[00:04:18] [ debug ] [brave]     mount.devfs
[00:04:18] [ debug ] [brave] Done.
[00:04:18] [ debug ] [brave] Locking brave ...
[00:04:18] [ info  ] [brave] Starting brave...
[00:04:18] [ debug ] [brave] Using `/usr/local/appjail/jails/brave/conf/template.conf` as the template.
[00:04:18] [ debug ] [brave] Writing `/usr/local/appjail/jails/brave/conf/template.conf` content to `/usr/local/appjail/cache/tmp/.appjail/appjail.NhMAfFVWky` ...
[00:04:18] [ debug ] [brave] Checking for parameters marked as required...
[00:04:18] [ debug ] [brave] Running: date +%Y-%m-%d.log
[00:04:18] [ debug ] [brave] exec.consolelog: /var/log/appjail/jails/brave/console/2023-06-30.log
[00:04:18] [ debug ] [brave] mount.fstab: /usr/local/appjail/jails/brave/conf/fstab
[00:04:18] [ debug ] [brave] host.hostname: brave.appjail
[00:04:18] [ debug ] [brave] Path: /usr/local/appjail/jails/brave/jail
[00:04:18] [ debug ] [brave] Resolving dependencies for brave...
[00:04:18] [ debug ] [brave] brave appended to the `seen` list.
[00:04:18] [ debug ] [brave] brave appended to the `resolved` list.
[00:04:18] [ debug ] [brave] Compiling template to `/usr/local/appjail/jails/brave/conf/jail.conf` ...
[00:04:18] [ debug ] [brave] jail.conf generated:
[00:04:18] [ debug ] [brave]     brave {
[00:04:18] [ debug ] [brave]         exec.start = "/bin/sh /etc/rc";
[00:04:18] [ debug ] [brave]         exec.stop = "/bin/sh /etc/rc.shutdown jail";
[00:04:18] [ debug ] [brave]         mount.devfs;
[00:04:18] [ debug ] [brave]         exec.consolelog = "/var/log/appjail/jails/brave/console/2023-06-30.log";
[00:04:18] [ debug ] [brave]         mount.fstab = "/usr/local/appjail/jails/brave/conf/fstab";
[00:04:18] [ debug ] [brave]         host.hostname = "brave.appjail";
[00:04:18] [ debug ] [brave]         path = "/usr/local/appjail/jails/brave/jail";
[00:04:18] [ debug ] [brave]     }
[00:04:18] [ debug ] [brave] Inspecting config.conf:
[00:04:18] [ debug ] [brave]     appjail_version: 2.5.2
[00:04:18] [ debug ] [brave]     birth: 1688049938
[00:04:18] [ debug ] [brave]     osarch: amd64
[00:04:18] [ debug ] [brave]     osversion: bionic
[00:04:18] [ debug ] [brave]     jail_type: linux+debootstrap
[00:04:18] [ debug ] [brave]     release_name: default
[00:04:18] [ debug ] [brave] Preparing a linuxjail...
[00:04:18] [ debug ] [brave] fdescfs kernel module already loaded.
[00:04:18] [ debug ] [brave] linprocfs kernel module already loaded.
[00:04:18] [ debug ] [brave] linsysfs kernel module already loaded.
[00:04:18] [ debug ] [brave] pty kernel module already loaded.
[00:04:18] [ debug ] [brave] tmpfs kernel module already loaded.
[00:04:18] [ debug ] [brave] linux kernel module already loaded.
[00:04:18] [ debug ] [brave] linux64 kernel module already loaded.
[00:04:18] [ debug ] [brave] Running initscript `/usr/local/appjail/jails/brave/init` ...
[00:04:18] [ debug ] [brave] `/usr/local/appjail/jails/brave/init` exits with status code 0
[00:04:18] [ debug ] [brave] Creating...
jail: brave: mount.fstab: /usr/local/appjail/jails/brave/jail//dev/shm: No such file or directory
[00:04:18] [ warn  ] [brave] An error has occurred while starting brave jail.
[00:04:18] [ warn  ] [brave] The brave jail will be stopped...
[00:04:18] [ warn  ] [brave] Running some counterparts in unattended mode...
[00:04:18] [ warn  ] [brave] brave is not running.
[00:04:18] [ error ] [brave] An error has occurred while starting brave jail.
[00:04:19] [ debug ] [brave] Unlocking brave ...
```

**Comment by DtxdF** [comment]
Hi @kenrap,

It looks like you are using Brave's Makejail instead of creating your own Makejail, that's why you don't have the `ruleset` argument.

**Makejail**:

```
INCLUDE options/network.makejail
INCLUDE gh+AppJail-makejails/brave

ARG ruleset=0

OPTION template=files/linux.conf
OPTION devfs_ruleset=${ruleset}
```

As you can see, Brave's Makejail is cloned by the personal Makejail using `INCLUDE gh+AppJail-makejails/brave`.

Follow the README for an explanation of the other options.

> jail: brave: mount.fstab: /usr/local/appjail/jails/brave/jail//dev/shm: No such file or directory

You need to put in your `/etc/devfs.rules` file the set of rules specified by the README and also reboot your system or - much better - restart `devfs`.

```sh
service devfs restart
```

Basically your ruleset has unhide `/dev/shm` and `/dev/shm/*`.

**Comment by DtxdF** [comment]
This issue has been closed due to inactivity. You can reopen it if you wish.
